ckanext/cfpb_extrafields/datastore_actions.py

```diff
@@ -10,52 +10,83 @@
 
 def default_datastore(rid):
     ''' provide default datastore parameter values and reduce duplication '''
-    defaults_tuple = collections.namedtuple('defaults_tuple', 'json_colname context data')
-    json_colname = 'jtable'
+    # WARNING: when create table is called, it must know all of the colnames and their order.
+    # Changing the field names WILL BREAK front end access to resources with the old names
+    # (adding new fields will not cause this problem).
+    title_colname = 'record_title' #title crosses wires with something in CKAN
+    subtitle_colname = 'record_subtitle'
+    datetime_colname = 'datetime'
+    response_colname = 'response'
+    response_timems_colname = 'response_timems'
+    json_colname = 'json_record'
+    all_fields =[{'id':title_colname,'type':'text'},
+                 {'id':subtitle_colname,'type':'text'},
+                 {'id':datetime_colname,'type':'timestamp'}, 
+                 {'id':response_colname,'type':'bool'}, 
+                 {'id':response_timems_colname,'type':'int'},
+                 {'id':json_colname,'type':'json'},]
     user = tk.get_action('get_site_user')({'ignore_auth': True}, {})
     context = {'user': user['name']}
     data = {'resource_id': rid}
-    return defaults_tuple(json_colname=json_colname, context=context, data=data)
-def create_datastore_json(rid, json_record, title_colname, json_title): 
-    ''' create a new datastore json element with primary key element of title_colname and record '''
+    defaults_tuple = collections.namedtuple('defaults_tuple', 'context data \
+    title_colname subtitle_colname json_colname datetime_colname response_colname \
+    response_timems_colname all_fields')
+    return defaults_tuple(context, data,
+                          title_colname, subtitle_colname, json_colname,
+                          datetime_colname, response_colname,
+                          response_timems_colname, all_fields)
+def create_datastore_placeholder(rid): 
+    ''' create a new datastore element with default colnames'''
     defaults = default_datastore(rid)
-    json_colname = defaults.json_colname
-    context = defaults.context
     data = defaults.data
     data.update({
-    'force':'true',
-    'fields': [{'id': title_colname},{'id': json_colname, 'type': 'json'}],
-    'records': [ {title_colname: json_title, json_colname: json_record} ]
+        'force':'true',
+        'fields': defaults.all_fields,
+        'records': []
     })
-    ds = tk.get_action('datastore_create')(context, data)
-def get_all_datastore_jsons(rid, title_colname, json_title): 
+    ds = tk.get_action('datastore_create')(defaults.context, data)
+
+# JSON related (for datadict):
+def create_datastore_json(rid, json_title, json_record): 
```

Comment:
Possibly merge this method with create_datastore_placeholder and use default values of "None" for json_title and json_record.